system/Commands/Translation/LocalizationFinder.php

```diff
@@ -0,0 +1,330 @@
+        $this->showNew      = array_key_exists('show-new', $params);
+        $optionLocale       = $params['locale'] ?? null;
+        $optionDir          = $params['dir'] ?? null;
+        $currentLocale      = Locale::getDefault();
+        $currentDir         = APPPATH;
+        $this->languagePath = $currentDir . 'Language';
+
+        if (ENVIRONMENT === 'testing') {
+            $currentDir         = SUPPORTPATH . 'Services/';
+            $this->languagePath = SUPPORTPATH . 'Language';
+        }
+
+        if (is_string($optionLocale)) {
+            if (! in_array($optionLocale, config(App::class)->supportedLocales, true)) {
+                CLI::error(
+                    'Error: "' . $optionLocale . '" is not supported. Supported locales: '
+                    . implode(', ', config(App::class)->supportedLocales)
+                );
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentLocale = $optionLocale;
+        }
+
+        if (is_string($optionDir)) {
+            $tempCurrentDir = realpath($currentDir . $optionDir);
+
+            if (false === $tempCurrentDir) {
+                CLI::error('Error: Directory must be located in "' . $currentDir . '"');
+
+                return EXIT_USER_INPUT;
+            }
+
+            if ($this->isSubDirectory($tempCurrentDir, $this->languagePath)) {
+                CLI::error('Error: Directory "' . $this->languagePath . '" restricted to scan.');
+
+                return EXIT_USER_INPUT;
+            }
+
+            $currentDir = $tempCurrentDir;
+        }
+
+        $this->process($currentDir, $currentLocale);
+
+        CLI::write('All operations done!');
+
+        return EXIT_SUCCESS;
+    }
+
+    /**
+     * @param SplFileInfo|string $file
+     */
+    private function findTranslationsInFile($file): array
+    {
+        $foundLanguageKeys = [];
+
+        if (is_string($file) && is_file($file)) {
+            $file = new SplFileInfo($file);
+        }
+
+        $fileContent = file_get_contents($file->getRealPath());
+        preg_match_all('/lang\(\'([._a-z0-9\-]+)\'\)/ui', $fileContent, $matches);
+
+        if ([] === $matches[1]) {
+            return [];
+        }
+
+        foreach ($matches[1] as $phraseKey) {
+            $phraseKeys = explode('.', $phraseKey);
+
+            /**
+             * Language key not have Filename or Lang key
+             */
+            if (count($phraseKeys) < 2) {
+                continue;
+            }
+
+            $languageFileName   = array_shift($phraseKeys);
+            $isEmptyNestedArray = (! empty($languageFileName) && empty($phraseKeys[0]))
+                || (empty($languageFileName) && ! empty($phraseKeys[0]))
+                || (empty($languageFileName) && empty($phraseKeys[0]));
+
+            if ($isEmptyNestedArray) {
+                continue;
+            }
+
+            /**
+             * Language key as string
+             */
```

Comment:
Please do not use doc block for comments. If single line, use `//`
```suggestion
             // language key as string
```

Reply:
I don't understand this comment. `$phraseKey` is still an array. How does it become a string?

Reply:
No. This is question:
```php
$phraseKey = 'App.title';
$phraseKeys = ['App', 'title'];
$filename = array_shift($phraseKeys);
$phraseKeys = ['title']; // count() return 1 
```

 "If count($phraseKeys) === 1 mean "Language key as string"
 It sounds a little different in my language. It's easier to delete a comment)